[CSS Zoom] Pass in zoom when evaluating padding during use time.

sammygill opened the pull request:
#### 10cdfef718dbe350058930473d7fcfe4beaba92e
<pre>
[CSS Zoom] Pass in zoom when evaluating padding during use time.
<a href="https://bugs.webkit.org/show_bug.cgi?id=301537">https://bugs.webkit.org/show_bug.cgi?id=301537</a>
<a href="https://rdar.apple.com/problem/163518723">rdar://problem/163518723</a>

Reviewed by Tim Nguyen and Lily Spiniolas.

This patch moves the PaddingEdge type to become Unzoomed. As part of
this process, any type we evaluate a PaddingEdge in the code we need to
supply a zoom factor. Most of this patch is passing in the used zoom
factor any time this happens.

* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style.html:
* Source/WebCore/layout/formattingContexts/FormattingGeometry.cpp:
(WebCore::Layout::FormattingGeometry::computedPadding const):
* Source/WebCore/layout/formattingContexts/block/BlockFormattingGeometry.cpp:
(WebCore::Layout::BlockFormattingGeometry::intrinsicWidthConstraints const):
* Source/WebCore/layout/formattingContexts/table/TableFormattingContext.cpp:
(WebCore::Layout::TableFormattingContext::computedPreferredWidthForColumns):
* Source/WebCore/layout/integration/LayoutIntegrationBoxGeometryUpdater.cpp:
(WebCore::LayoutIntegration::usedValueOrZero):
(WebCore::LayoutIntegration::BoxGeometryUpdater::logicalPadding):
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::borderMarginOrPaddingWidth):
(WebCore::getBorderPaddingMargin):
* Source/WebCore/rendering/RenderBoxModelObjectInlines.h:
(WebCore::RenderBoxModelObject::computedCSSPaddingAfter const):
(WebCore::RenderBoxModelObject::computedCSSPaddingBefore const):
(WebCore::RenderBoxModelObject::computedCSSPaddingBottom const):
(WebCore::RenderBoxModelObject::computedCSSPaddingEnd const):
(WebCore::RenderBoxModelObject::computedCSSPaddingLeft const):
(WebCore::RenderBoxModelObject::computedCSSPaddingRight const):
(WebCore::RenderBoxModelObject::computedCSSPaddingStart const):
(WebCore::RenderBoxModelObject::computedCSSPaddingTop const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::popupInternalPaddingBox const):
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
Much of the form control code computes the padding based off of the em
size, which is an already zoomed value since font size is zoomed.

One option would be to force the code which computes the em size to use the
&quot;unzoomedComputedSize,&quot; on FontDescription by using the appropriate
conversion data. However, this is not ideal because we may get a
different value than what we previous got. For example, a font size of
11px with a used zoom of 3 would previously give us a value of 16px, after
some rounding when computing a 1em size. If we took the approach I just
described, this would give us a size of 5px, which would then get
evaluated to 15px.

In order to maintain behavior, I opted to take a different approach.
Instead, we continue to compute the padding sizes in the same way we
were previously, but instead divide out the used zoom right before we apply
it to the RenderStyle. This gives allows us to continue to get the same
values we were previously computing while also still using the correct
size during use time.

(WebCore::applyEmPadding):
(WebCore::applyCommonButtonPaddingToStyleForVectorBasedControls):
(WebCore::RenderThemeCocoa::adjustButtonStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::paintMenuListButtonDecorationsForVectorBasedControls):
(WebCore::RenderThemeCocoa::adjustSearchFieldCancelButtonStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::adjustedMaximumLogicalWidthForControl const):
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::popupInternalPaddingBox const):
(WebCore::RenderThemeIOS::paintMenuListButtonDecorations):
(WebCore::RenderThemeIOS::paintListButton):
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::popupInternalPaddingBox const):
* Source/WebCore/style/values/box/StylePadding.h:

Canonical link: <a href="https://commits.webkit.org/302229@main">https://commits.webkit.org/302229@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2e9580ff1562ca34a9bc221d7b7dfb61ac59805c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/128430 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/700 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/39261 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/135824 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/79879 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/c63da999-5436-4df4-9fc0-13df9325a615) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/130302 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/643 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/573 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/97761 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/65662 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f28129e9-f04e-4424-ad1a-9660ddc8d388) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/131378 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/455 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115063 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/78371 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-26-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33170 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79107 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/120444 "Built successfully and passed tests") | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/33654 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/138274 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/126890 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/541 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/510 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106303 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/581 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/111404 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106114 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27027 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/448 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/29947 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/52863 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/592 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/63782 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/159914 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/487 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39940 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/546 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/550 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Can you add entries for all padding props to the svg-computed-style.html test?

This would also ideally have a rendering test though it could be a follow up.

Also tagging @alanbaradlay in case he has any thoughts.

lilyspiniolas commented on Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
```diff
@@ -350,7 +350,7 @@ Style::PaddingBox RenderThemeAdwaita::popupInternalPaddingBox(const RenderStyle&
         return { 0_css_px };
 
-    auto zoomedArrowSize = menuListButtonArrowSize * style.usedZoom();
+    auto zoomedArrowSize = menuListButtonArrowSize;
```
No need to create `zoomedArrowSize` anymore

lilyspiniolas reviewed (approved):
nit: typo in commit message "before apply it to the RenderStyle"

sammygill committed:
[CSS Zoom] Pass in zoom when evaluating padding during use time.
https://bugs.webkit.org/show_bug.cgi?id=301537
rdar://problem/163518723

Reviewed by Tim Nguyen and Lily Spiniolas.

This patch moves the PaddingEdge type to become Unzoomed. As part of
this process, any type we evaluate a PaddingEdge in the code we need to
supply a zoom factor. Most of this patch is passing in the used zoom
factor any time this happens.

* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style.html:
* Source/WebCore/layout/formattingContexts/FormattingGeometry.cpp:
(WebCore::Layout::FormattingGeometry::computedPadding const):
* Source/WebCore/layout/formattingContexts/block/BlockFormattingGeometry.cpp:
(WebCore::Layout::BlockFormattingGeometry::intrinsicWidthConstraints const):
* Source/WebCore/layout/formattingContexts/table/TableFormattingContext.cpp:
(WebCore::Layout::TableFormattingContext::computedPreferredWidthForColumns):
* Source/WebCore/layout/integration/LayoutIntegrationBoxGeometryUpdater.cpp:
(WebCore::LayoutIntegration::usedValueOrZero):
(WebCore::LayoutIntegration::BoxGeometryUpdater::logicalPadding):
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::borderMarginOrPaddingWidth):
(WebCore::getBorderPaddingMargin):
* Source/WebCore/rendering/RenderBoxModelObjectInlines.h:
(WebCore::RenderBoxModelObject::computedCSSPaddingAfter const):
(WebCore::RenderBoxModelObject::computedCSSPaddingBefore const):
(WebCore::RenderBoxModelObject::computedCSSPaddingBottom const):
(WebCore::RenderBoxModelObject::computedCSSPaddingEnd const):
(WebCore::RenderBoxModelObject::computedCSSPaddingLeft const):
(WebCore::RenderBoxModelObject::computedCSSPaddingRight const):
(WebCore::RenderBoxModelObject::computedCSSPaddingStart const):
(WebCore::RenderBoxModelObject::computedCSSPaddingTop const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::popupInternalPaddingBox const):
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
Much of the form control code computes the padding based off of the em
size, which is an already zoomed value since font size is zoomed.

One option would be to force the code which computes the em size to use the
"unzoomedComputedSize," on FontDescription by using the appropriate
conversion data. However, this is not ideal because we may get a
different value than what we previous got. For example, a font size of
11px with a used zoom of 3 would previously give us a value of 16px, after
some rounding when computing a 1em size. If we took the approach I just
described, this would give us a size of 5px, which would then get
evaluated to 15px.

In order to maintain behavior, I opted to take a different approach.
Instead, we continue to compute the padding sizes in the same way we
were previously, but instead divide out the used zoom right before we apply
it to the RenderStyle. This gives allows us to continue to get the same
values we were previously computing while also still using the correct
size during use time.

(WebCore::applyEmPadding):
(WebCore::applyCommonButtonPaddingToStyleForVectorBasedControls):
(WebCore::RenderThemeCocoa::adjustButtonStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::paintMenuListButtonDecorationsForVectorBasedControls):
(WebCore::RenderThemeCocoa::adjustSearchFieldCancelButtonStyleForVectorBasedControls const):
(WebCore::RenderThemeCocoa::adjustedMaximumLogicalWidthForControl const):
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::popupInternalPaddingBox const):
(WebCore::RenderThemeIOS::paintMenuListButtonDecorations):
(WebCore::RenderThemeIOS::paintListButton):
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::popupInternalPaddingBox const):
* Source/WebCore/style/values/box/StylePadding.h:

Canonical link: https://commits.webkit.org/302229@main
